src/gar.lua

```diff
@@ -756,4 +778,61 @@ function gar.cancelDelegateWithdrawal(from, gatewayAddress, vaultId)
 	}
 end
 
+function gar.instantDelegateWithdrawal(from, gatewayAddress, vaultId, currentTimestamp)
+	local gateway = gar.getGateway(gatewayAddress)
+	if gateway == nil then
+		error("Gateway does not exist")
+	end
+
+	local delegate = gateway.delegates[from]
+	if delegate == nil then
+		error("Delegate does not exist")
+	end
+
+	local vault = delegate.vaults[vaultId]
+	if vault == nil then
+		error("Vault does not exist")
+	end
+
+	-- Calculate elapsed time since the withdrawal started
+	local elapsedTime = currentTimestamp - vault.startTimestamp
+	local totalWithdrawalTime = gar.getSettings().delegates.withdrawLengthMs
+
+	-- Ensure the elapsed time is not negative
+	if elapsedTime < 0 then
+		error("Invalid elapsed time")
+	end
+
+	-- Calculate the penalty rate based on elapsed time
+	local maxPenalty = 0.80
+	local minPenalty = 0.05
+	local penaltyRate = maxPenalty - ((maxPenalty - minPenalty) * (elapsedTime / totalWithdrawalTime))
```

Comment:
nit: use the constants you added

Reply:
Done.

https://github.com/ar-io/ar-io-network-process/pull/97/commits/0f8b37a3d29046fead347d62fe2cf4f9f146c711